### PR TITLE
Code completion

### PR DESCRIPTION
**Pull request by Wzixiao**
这是一个大的 pr，关于续写功能的所有 pr 均往此 branch 中 merge

### CODE REVIEW

**Review by liyongsea — COMMENTED** [review]
continue writing => code complement

File: packages/jupyter-ai/src/index.ts [hunk 1]
```diff
@@ -49,13 +51,29 @@ const plugin: JupyterFrontEndPlugin<void> = {
...
 
+    /**
+     * Initialize bigcode settings widget
+     */
+    const bigcodeWidget = buildBigcodeSidebar();
+
+    /**
+     * Initialize keydown handler
+     */
+    handleCodeCompletionKeyDown(app);
```
Comment (liyongsea) [inline packages/jupyter-ai/src/index.ts:62]: 不要随时监听，用户激活了以后再监听

File: packages/jupyter-ai/src/keydown-handler.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+const mountExtension = (
+  editor: CodeMirrorEditor,
+  extension: Extension
+): void => {
+  // If the editor has already been processed
+  if (mountedEditors.has(editor)) {
+    return;
+  }
+
+  // editor 还没有正常的完成初始化
```
Comment (liyongsea) [inline packages/jupyter-ai/src/keydown-handler.ts]: english

File: packages/jupyter-ai/src/utils/bigcode-request.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import { ICell } from '../types/cell';
+
+/**
+ * Generates the appropriate prompt string based on the cell type.
+ * The cell type can be either 'code' or 'markdown'.
+ *
+ * @param {ICell} cell - The cell object which includes the type and content.
+ * @returns {string} The generated prompt string for the cell.
+ */
+const getPromptForCell = (cell: ICell): string => {
```
Comment (liyongsea) [inline packages/jupyter-ai/src/utils/bigcode-request.ts:11]: 感觉output没有放进去

Comment (liyongsea) [reply packages/jupyter-ai/src/utils/bigcode-request.ts:11]: <start_jupyter><jupyter_text>text<jupyter_code>code<jupyter_output>output<jupyter_text>

File: packages/jupyter-ai/src/keydown-handler.ts (hunk 2 above)
Comment (Wzixiao) [reply packages/jupyter-ai/src/keydown-handler.ts]: solved https://github.com/Wzixiao/jupyter-ai-bigcode-code-completion/pull/23

File: packages/jupyter-ai/src/utils/bigcode-request.ts (hunk 3 above)
Comment (Wzixiao) [reply packages/jupyter-ai/src/utils/bigcode-request.ts:11]: solved  https://github.com/Wzixiao/jupyter-ai-bigcode-code-completion/pull/24

**Pull request merged**